src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  onDeleteTodo?: (todoId: number) => void;
+  loadingTodoId?: number | null;
+  onUpdateTodo?: (todoToUpdate: Todo, newTitle?: string) => void;
+}
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  onDeleteTodo,
+  loadingTodoId = null,
+  onUpdateTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const { id, title, completed } = todo;
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (loadingTodoId === id) {
+      setIsLoading(true);
+    } else {
+      setIsLoading(false);
+    }
+  }, [loadingTodoId]);
+
+  useEffect(() => {
+    if (isEditingTitle) {
+      inputRef.current?.focus();
+    }
+  }, [isEditingTitle]);
+
+  const handleDeleteTodo = () => (
+    onDeleteTodo && onDeleteTodo(id)
+  );
+
+  const handleUpdateStatus = () => (
+    onUpdateTodo && onUpdateTodo(todo)
```

Comment:
I think better would be to make these functions required and for the tempTodo create another simplified component